controllers/applicationvolumereplication_controller.go

```diff
@@ -538,40 +585,137 @@ func (r *ApplicationVolumeReplicationReconciler) processUnpausedSubscription(
 	}
 
 	if err := r.createOrUpdateVRGManifestWork(
-		subscription.Name, subscription.Namespace, homeCluster,
-		avr.Spec.S3Endpoint, avr.Spec.S3SecretName); err != nil {
+		subscription.Name, subscription.Namespace,
+		homeCluster, avr.Spec.S3Endpoint, avr.Spec.S3SecretName, avr.Spec.PVCSelector); err != nil {
 		r.Log.Error(err, "failed to create or update VolumeReplicationGroup manifest")
 
 		return ramendrv1alpha1.SubscriptionPlacementDecision{}, err
 	}
 
+	prevHomeCluster := ""
+
+	if d, found := avr.Status.Decisions[subscription.Name]; found {
+		switch {
+		case d.PrevHomeCluster == "":
+			prevHomeCluster = d.HomeCluster
+		case d.PrevHomeCluster == homeCluster:
+			prevHomeCluster = ""
+		}
+	}
+
 	return ramendrv1alpha1.SubscriptionPlacementDecision{
-		HomeCluster: homeCluster,
-		PeerCluster: peerCluster,
+		HomeCluster:     homeCluster,
+		PeerCluster:     peerCluster,
+		PrevHomeCluster: prevHomeCluster,
 	}, nil
 }
 
-func (r *ApplicationVolumeReplicationReconciler) findNextHomeCluster(
+func (r *ApplicationVolumeReplicationReconciler) findNextHomeClusterFromPlacementRule(
+	subscription *subv1.Subscription) (string, error) {
+	r.Log.Info("Finding the next home cluster for subscription", "name", subscription.Name)
+
+	placementRule, err := r.getPlacementRule(subscription)
+	if err != nil {
+		return "", err
+	}
+
+	if len(placementRule.Status.Decisions) == 0 {
+		return "", fmt.Errorf("no decisions were found in placementRule %s", placementRule.Name)
+	}
+
+	const requiredClusterReplicas = 1
+
+	if placementRule.Spec.ClusterReplicas != nil && *placementRule.Spec.ClusterReplicas != requiredClusterReplicas {
+		return "", fmt.Errorf("PlacementRule %s Required cluster replicas %d != %d",
+			placementRule.Name, requiredClusterReplicas, *placementRule.Spec.ClusterReplicas)
+	}
+
+	return placementRule.Status.Decisions[0].ClusterName, nil
+}
+
+func (r *ApplicationVolumeReplicationReconciler) validateHomeClusterSelection(
+	avr *ramendrv1alpha1.ApplicationVolumeReplication,
+	subscription *subv1.Subscription, newHomeCluster string) (string, error) {
+	if _, found := avr.Status.Decisions[subscription.Name]; !found {
+		return newHomeCluster, nil
+	}
+
+	if action, found := avr.Spec.DREnabledSubscriptions[subscription.Name]; found {
```

Comment:
If we do not find the subscription we return no error (line 654), so even if user did not mention subscription in the map we would failover/failback this subscription?